### PR TITLE
Update README (and JSON metadata file)

### PR DESCRIPTION
**Pull request by kbrandl**
Several updates to the README:
- updated intro section
- updated instructions in the **Using this project** section and **Making changes** section to reflect current contents of sample project and current process for testing changes to the project
- updated the **Debugging** section to replace detailed info with pointer to the docs
- removed the **Intellisense** section (since these steps are no longer necessary)
- updated the **Questions and comments** section to indicate correct Stack Overflow tag for Office Add-ins questions
- updated the **Additional resources** section to add a link to the custom functions tutorial and to filter the list of repos shown by the `github.com/OfficeDev` link

Also, changed the `helpUrl` in **customfunctions.json** to use "contoso.com" (instead of "dev.office.com").

### CODE REVIEW

**Comment by kbrandl** [comment]
@Zlatkovsky this PR cleans up the README as we discussed recently. Can you please review and provide feedback when you have a minute?  (After this PR is reviewed/approved and merged to **master**, I'll submit the same README updates to the `preview-sept-2018-javascript` branch and the `preview-sept-2018-typescript` branch of this repo.)

File: README.md [hunk 1]
```diff
@@ -1,78 +1,76 @@
...
-* [To use the project](#to-use-the-project)
+* [Using this project](#using-this-project)
 * [Making changes](#making-changes)
 * [Debugging](#debugging)
-* [IntelliSense for the JSON file in Visual Studio Code](#intellisense-for-the-json-file-in-visual-studio-code)
 * [Questions and comments](#questions-and-comments)
 * [Additional resources](#additional-resources)
 
-## Change History
+## Change history
```
Comment (akrantz) [inline README.md]: Do we need the "Change history" section? Or perhaps put it at then end after "Additional resources"...

Comment (kbrandl) [reply README.md]: @akrantz thanks for raising this question. IMO, an out-of-date **Change history** section is worse than having no **Change history** section at all.  (I almost removed it during my first pass of README updates.) Given that nobody's been maintaining this section, I'm going to remove it for now; someone can always add it back later if they feel strongly about updating/maintaining change history info going forward.

**Comment by kbrandl** [comment]
@Zlatkovsky given @akrantz's approval (and the relative insignificance of README updates), I'm going to go ahead and merge this PR to master. Please do feel free to reach out with any feedback you may have though; we can always incorp feedback into the next round of README updates (at some point in the future).

**Pull request merged**